src/components/DopplerPlus/UpgradeSuggestionForm.test.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import UpgradeSuggestionForm from './UpgradeSuggestionForm';
+import { AppServicesProvider } from '../../services/pure-di';
+import DopplerIntlProvider from '../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import { act } from 'react-dom/test-utils';
+
+describe('UpgradeSuggestionForm component', () => {
+  afterEach(cleanup);
+
+  const dependencies = {
+    appSessionRef: {
+      current: {
+        userData: {
+          user: {
+            email: 'hardcoded@email.com',
+          },
+        },
+      },
+    },
+    dopplerLegacyClient: {
+      requestSuggestionUpgradeForm: async () => {
+        return { success: true };
+      },
+    },
+    captchaUtilsService: {
+      useCaptcha: () => {
+        const Captcha = () => null;
+        const verifyCaptcha = async () => {
+          return { success: true, captchaResponseToken: 'hardcodedResponseToken' };
+        };
+        const recaptchaRef = null;
+        return [Captcha, verifyCaptcha, recaptchaRef];
+      },
+    },
+  };
+
+  const UpgradeSuggestionFormElement = () => (
+    <AppServicesProvider forcedServices={dependencies}>
+      <DopplerIntlProvider>
+        <UpgradeSuggestionForm />
+      </DopplerIntlProvider>
+    </AppServicesProvider>
+  );
+  jest.useFakeTimers();
+
+  it('should show success message if submit succesfully', async () => {
+    // Arrange
+    const { container, getByText } = render(<UpgradeSuggestionFormElement />);
+
+    // Act
+    act(() => {
+      const inputName = container.querySelector('input#name');
+      fireEvent.change(inputName, { target: { value: 'Juan' } });
+
+      const inputLastname = container.querySelector('input#lastname');
+      fireEvent.change(inputLastname, { target: { value: 'Perez' } });
+
+      const inputPhone = container.querySelector('input#phone');
+      fireEvent.change(inputPhone, { target: { value: '+54 223 655-8877' } });
+    });
+
+    act(() => {
+      const submitButton = container.querySelector('button[type="submit"]');
+      fireEvent.submit(submitButton);
+    });
```

Comment:
Perhaps the waitFor should be here instead of the expect. @jcamposmk what do you think?

Reply:
maybe you can take a look at this @silsanchez8  https://kentcdodds.com/blog/common-mistakes-with-react-testing-library
I have learned a lot about testing there